diff --git a/3rdparty/protobuf/src/google/protobuf/io/coded_stream.h b/3rdparty/protobuf/src/google/protobuf/io/coded_stream.h
index 45963ef86b..031ac57a57 100644
--- a/3rdparty/protobuf/src/google/protobuf/io/coded_stream.h
+++ b/3rdparty/protobuf/src/google/protobuf/io/coded_stream.h
@@ -132,7 +132,7 @@
 #pragma runtime_checks("c", off)
 #endif
 #else
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__vita__)
 #include <machine/endian.h>  // __BYTE_ORDER
 #elif defined(__FreeBSD__)
 #include <sys/endian.h>  // __BYTE_ORDER
diff --git a/3rdparty/protobuf/src/google/protobuf/stubs/mutex.h b/3rdparty/protobuf/src/google/protobuf/stubs/mutex.h
index c4599913be..f773ce7db1 100644
--- a/3rdparty/protobuf/src/google/protobuf/stubs/mutex.h
+++ b/3rdparty/protobuf/src/google/protobuf/stubs/mutex.h
@@ -118,7 +118,7 @@ class CallOnceInitializedMutex {
 // mutexes.
 class GOOGLE_PROTOBUF_CAPABILITY("mutex") PROTOBUF_EXPORT WrappedMutex {
  public:
-#if defined(__QNX__)
+#if defined(__QNX__) || defined(__vita__)
   constexpr WrappedMutex() = default;
 #else
   constexpr WrappedMutex() {}
diff --git a/3rdparty/protobuf/src/google/protobuf/stubs/port.h b/3rdparty/protobuf/src/google/protobuf/stubs/port.h
index 045e25d892..2872930d8f 100644
--- a/3rdparty/protobuf/src/google/protobuf/stubs/port.h
+++ b/3rdparty/protobuf/src/google/protobuf/stubs/port.h
@@ -57,7 +57,7 @@
 #pragma runtime_checks("c", off)
 #endif
 #else
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__vita__)
 #include <machine/endian.h>  // __BYTE_ORDER
 #elif defined(__FreeBSD__)
 #include <sys/endian.h>  // __BYTE_ORDER
diff --git a/modules/core/src/system.cpp b/modules/core/src/system.cpp
index 027072a5da..a694e53e8a 100644
--- a/modules/core/src/system.cpp
+++ b/modules/core/src/system.cpp
@@ -655,6 +655,9 @@ struct HWFeatures
     #ifdef __mips_msa
         have[CV_CPU_MSA] = true;
     #endif
+    #ifdef __vita__
+        have[CV_CPU_NEON] = true;
+    #endif
 
     #if (defined __ppc64__ || defined __PPC64__) && defined __linux__
         unsigned int hwcap = getauxval(AT_HWCAP);
diff --git a/modules/dnn/src/torch/THDiskFile.cpp b/modules/dnn/src/torch/THDiskFile.cpp
index 84b6b23e81..b335f0c522 100644
--- a/modules/dnn/src/torch/THDiskFile.cpp
+++ b/modules/dnn/src/torch/THDiskFile.cpp
@@ -112,7 +112,7 @@ static void THDiskFile_seek(THFile *self, long position)
 
 #if defined(_WIN64)
   if(_fseeki64(dfself->handle, (__int64)position, SEEK_SET) < 0)
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__vita__)
   if(fseek(dfself->handle, (long)position, SEEK_SET) < 0)
 #else
   if(fseeko(dfself->handle, (off_t)position, SEEK_SET) < 0)
@@ -132,7 +132,7 @@ static void THDiskFile_seekEnd(THFile *self)
 
 #if defined(_WIN64)
   if(_fseeki64(dfself->handle, 0L, SEEK_END) < 0)
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__vita__)
   if(fseek(dfself->handle, 0L, SEEK_END) < 0)
 #else
   if(fseeko(dfself->handle, 0L, SEEK_END) < 0)
@@ -151,7 +151,7 @@ static long THDiskFile_position(THFile *self)
 
 #if defined(_WIN64)
   __int64 offset = _ftelli64(dfself->handle);
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__vita__)
   long offset = ftell(dfself->handle);
 #else
   off_t offset = ftello(dfself->handle);
